### PR TITLE
Check got_int in msg_multiline_attr with ex_echo

### LINKED ISSUE #10923: "-- More --" prompt cannot be quit using "q" (:echon)
Given the command:
```vim
:echon join(map(range(0, &lines*3), 'v:val'), "\n")
```

Running `nvim -u NORC` and executing it will display a screen full of numbers, and a "-- More --" prompt.

The more prompt cannot be quit anymore after 67bac681e ("ex_echo: reuse code from message.c to show arg to user"): pressing "q" will display the next page, just like "<Enter>".
Before 67bac681e it would quit the pager (as it happens in Vim).

/cc @bfredl @AlexandreDubray

### PR DESCRIPTION
**Pull request by blueyed**
Fixes quitting the pager using `q`.

Fixes https://github.com/neovim/neovim/issues/10923.

### CODE REVIEW

**Comment by blueyed** [comment]
Without the fix the test would still pass, but hang then afterwards.

1. I think the 2nd `screen:expect` should contain the more prompt then still
2. The whole test suite should not hang when a test ends with a hit-enter prompt

**Comment by bfredl** [comment]
> The whole test suite should not hang when a test ends with a hit-enter prompt

This is only because of `after_each(function() screen:detach() end)`. This is not necessary and can be removed in most places.

**Comment by bfredl** [comment]
> I think the 2nd screen:expect should contain the more prompt then still

Need `command("set more")` setup for this.

**Comment by blueyed** [comment]
> Need `command("set more")` setup for this.

Sure, fixed.

File: test/functional/ui/screen_basic_spec.lua [hunk 1]
```diff
@@ -951,6 +951,46 @@ local function screen_tests(linegrid)
                                                            |
     ]])
   end)
+
+  describe('more prompt', function()
```
Comment (bfredl) [inline test/functional/ui/screen_basic_spec.lua]: I don't think this belong in `screen_basic_spec.lua`. Perhaps a "pager" section of `ui/messages_spec.lua` . There will be more pager tests with #10930 .

Comment (blueyed) [reply test/functional/ui/screen_basic_spec.lua]: Ok, moved.

File: src/nvim/message.c [hunk 2]
```diff
@@ -227,7 +227,7 @@ void msg_multiline_attr(const char *s, int attr)
 {
   const char *next_spec = s;
 
-  while (next_spec != NULL) {
+  while (next_spec != NULL && !got_int) {
```
Comment (bfredl) [inline src/nvim/message.c]: Perhaps this is too general. The previous implementation used `got_int` only in `ex_echo`. But this function is also uses for certain error messages (like lua errors), seems there is a risk that such an error message caused by an interrupt can get inhibited.

 `msg_multiline_attr` could take a `bool` flag which is only set to true for the usage in `ex_echo`, to follow the previously correct implementation more closely.

Comment (blueyed) [reply src/nvim/message.c]: ok, done.

**Review by bfredl — APPROVED** [review]

**Review by justinmk — APPROVED** [review]

**Comment by bfredl** [comment]
Weird appveyor error, assumed unrelated:
```
 [  ERROR   ] 1 error, listed below:
  [  ERROR   ] test/functional\ui\wildmode_spec.lua @ 84: 'wildmenu' is preserved during :terminal activity
  test\functional\helpers.lua:591: crash? request failed
  
  stack traceback:
  	test\functional\helpers.lua:591: in function 'assert_alive'
  	test\functional\helpers.lua:302: in function 'nvim_feed'
  	test\functional\helpers.lua:311: in function 'feed'
  	test/functional\ui\wildmode_spec.lua:112: in function <test/functional\ui\wildmode_spec.lua:84>

```

**Comment by justinmk** [comment]
@bfredl That failure started since https://github.com/neovim/neovim/commit/a00eb23c27da7a0b79ea0030370074a1e383e457

**Comment by blueyed** [comment]
@bfredl 
JFI: should have been squash-merged.

**Pull request merged**